### ISSUE TITLE
Long-lived peer connection went down, netaddr is null

### ISSUE THREAD

**Issue by bleetube** [ElementsProject/lightning#6149]
A week or so ago I lost connectivity to just one peer (a popular liquidity provider) that I've been connected to for months. The peer's node is definitely online, so I investigated.  I noticed that there is no netaddr associated with that peer in listpeers. I've tried lightning-cli connect with either their clearnet or tor address, but their netaddr is still null. Maybe `netaddr:null` is a red herring, but my spidey (bug) senses tingled, so I'm sharing my experience. I should also mention that I intentionally do not publish any address, even via tor. Kind of a lame experiment with privacy+defense in depth.

Basic connectivity test:

```bash
$ torsocks nc -vz ecu3omnk6kxer5hw35owlzhw3xuqfroxjnnflbkjkc7xy2jy3gy7b2yd.onion 9735
ecu3omnk6kxer5hw35owlzhw3xuqfroxjnnflbkjkc7xy2jy3gy7b2yd.onion [127.42.42.0] 9735 open
```

`$ lightning-cli connect 024bfaf0cabe7f874fd33ebf7c6f4e5385971fc504ef3f492432e9e3ec77e1b5cf ecu3omnk6kxer5hw35owlzhw3xuqfroxjnnflbkjkc7xy2jy3gy7b2yd.onion 9735`
```json
{
   "id": "024bfaf0cabe7f874fd33ebf7c6f4e5385971fc504ef3f492432e9e3ec77e1b5cf",
   "features": "800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000888a52a1",
   "direction": "out",
   "address": {
      "type": "torv3",
      "address": "ecu3omnk6kxer5hw35owlzhw3xuqfroxjnnflbkjkc7xy2jy3gy7b2yd.onion",
      "port": 9735
   }
}
```

`$ lightning-cli listpeers | jq '.peers[] | {id, connected, netaddr}'`:
```json
{
  "id": "024bfaf0cabe7f874fd33ebf7c6f4e5385971fc504ef3f492432e9e3ec77e1b5cf",
  "connected": false,
  "netaddr": null
}
{
  "id": "03d83d26813669d8665d6a5017ecd18d176db0893f00f39e0a38493a1b6e3bbcd7",
  "connected": true,
  "netaddr": [
    "h5kduggrox57j2lz6457he3dfbp7o4tmn2w2jdett3fkbonw42ikbiad.onion:9735"
  ]
}
```
(other peers are connected just fine)

lightningd log:
```syslog
024bfa...-chan#1: Peer transient failure in CHANNELD_NORMAL: Disconnected
024bfa...-chan#2: Peer transient failure in CHANNELD_NORMAL: Disconnected
024bfa...-lightningd: Unknown channel <redacted> for WIRE_CHANNEL_REESTABLISH
``` 

### `getinfo` output

```json
{
   "id": "<redacted>",
   "alias": "<redacted>",
   "color": "<redacted>",
   "num_peers": "<redacted>",
   "num_pending_channels": 0,
   "num_active_channels": "<redacted>",
   "num_inactive_channels": 0,
   "address": [],
   "binding": [
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 9735
      }
   ],
   "version": "v23.02.2-modded",
   "blockheight": 783969,
   "network": "bitcoin",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/home/lightning/.lightning/bitcoin",
   "our_features": {
      "init": "08a0882a2269a2",
      "node": "88a0882a2269a2",
      "channel": "",
      "invoice": "02000022024100"
   }
}
```
I build with `--enable-experimental-features` and use:
* experimental-onion-messages
* experimental-offers
* experimental-dual-fund
* fetchinvoice-noconnect
And two plugins: 
* prometheus.py (prometheus metric exporter)
* clrest.js (c-lightning-REST)

**Comment by bleetube** [comment]
300,000 sats the person that can help me sort this out

**Comment by bleetube** [comment]
@endothermicdev I was asked to tag you on this by nifty because it sounds like a gossip bug

**Comment by rustyrussell** [comment]
OK, netaddr isn't null, it's just not present at all for disconnected peers. jq is trying to tell you that.

Your real problem: `024bfa...-lightningd: Unknown channel <redacted> for WIRE_CHANNEL_REESTABLISH` means it is trying to reestablish a channel we know nothing about.  We will hang up at that point.

Any trace of that channel in yours logs?

**Comment by bleetube** [comment]
OH! The WIRE_CHANNEL_REESTABLISH messages only refer to my newest (third) channel with this peer, not the other two that have been open and active for a few months.

As it happens, this 3rd channel was stuck in the mempool for a couple weeks until I contacted this peer (the liquidity provider). He graciously did a CPFP tx to bump and get it confirmed back on 3/30.  But I never saw that channel as in my lightningd channel list. Turns out I should have just looked at my systemd journal.

My logs show that within two minutes of this tx confirming onchain, the first two channels started reporting "Peer connection lost" and "Peer transient failure."  It's all starting to make sense now.

Looking further back in the log, I saw this `Forgetting channel` message two days earlier on 3/28:

`UNUSUAL <redact>-chan#3: Forgetting channel: It has been 2016 blocks without the funding transaction <redact> getting deeply confirmed. We are fundee and can forget channel without loss of funds.`

So in a nutshell, if I'm right, it looks like my node thinks there are only two channels, but my peer knows there is three channels. The two nodes are thus failing to agree.

I'm guessing the quick fix would be to just close the 3rd channel (I'd have to ask the peer to do it). Is there a way to tell my node about the third channel and recover without closing a channel?

**Comment by rustyrussell** [comment]
Ah!!!  Right!  It's their money, so your node gave up after 2 weeks since it didn't have any funds at risk.

You'll have to ask them to close, sorry :(

The inability to hold a connection is due to us hanging up every time we send a warning: I'm going to leave this issue open so I can think about that (for next release!).

**Comment by kallerosenbaum** [comment]
It seems I'm in the same situation
```
2021-01-23T04:11:12.288Z UNUSUAL <CHANX>-chan#39645: Forgetting channel: It has been 2016 blocks without the funding transaction <TXIDA> getting deeply confirmed. We are fundee and can forget channel without loss of funds.
...
2023-10-25T20:04:36.741Z UNUSUAL X-lightningd: Unknown channel <TXIDA-BYTEREVERSED> for WIRE_CHANNEL_REESTABLISH
```
As you can see, it's been going on for years...
Let me know if I should look something something up.

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: About deprecated feature

If you believe this issue should remain open, please comment and we can reopen it.
